docs/portal.md

```diff
@@ -459,6 +459,80 @@ for vocabulary_name in common_vocabularies:
     assert vocabulary_name in vocabulary_names
 ```
 
+(portal-add-catalog-indexes-example)=
+
+## Add catalog indexes
+
+To add new indexes to the portal catalog if they don't already exist, use {meth}`api.portal.add_catalog_indexes`.
+
+```python
+from plone import api
+
+# Add new indexes with default logging
+indexes = [
+    ('custom_field', 'FieldIndex'),
+    ('review_date', 'DateIndex')
+]
+api.portal.add_catalog_indexes(wanted_indexes=indexes)
+
+# Add indexes but skip reindexing
+api.portal.add_catalog_indexes(
+    wanted_indexes=[('modified_by', 'FieldIndex')],
+    reindex=False
+)
+
+# Add indexes with custom logger
+import logging
+custom_logger = logging.getLogger('my.package')
+api.portal.add_catalog_indexes(
+    wanted_indexes=[('creation_user', 'FieldIndex')],
+    logger=custom_logger
+)
+```
+
+% invisible-code-block: python
+%
+% # Verify the indexes were added to the catalog
+% catalog = api.portal.get_tool('portal_catalog')
+% self.assertIn('custom_field', catalog.indexes())
+% self.assertIn('review_date', catalog.indexes())
+% self.assertIn('modified_by', catalog.indexes())
+% self.assertIn('creation_user', catalog.indexes())
+
+This function returns a list of the names of the indexes that were added.
+
+(portal-add-catalog-metadata-example)=
+
+## Add catalog metadata columns
+
+To add new metadata columns to the portal catalog if they don't already exist, use {meth}`api.portal.add_catalog_metadata`.
```

Comment:
````suggestion
To add metadata columns to the portal catalog, use {meth}`api.portal.add_catalog_metadata`.
This function returns a list of the names of the columns that were added.

```{note}
Adding metadata columns only makes them available for storage.
You still need to reindex your content to populate the values.
```

The following collection of code snippets adds metadata columns with either the default logger or a custom logger.
````